sklearn/feature_extraction/text.py

```diff
@@ -1056,18 +1066,19 @@ def transform(self, X, copy=True):
         -------
         vectors : sparse matrix, [n_samples, n_features]
         """
-        if hasattr(X, 'dtype') and np.issubdtype(X.dtype, np.float):
-            # preserve float family dtype
-            X = sp.csr_matrix(X, copy=copy)
-        else:
-            # convert counts or binary occurrences to floats
-            X = sp.csr_matrix(X, dtype=np.float64, copy=copy)
+        X = check_array(X, accept_sparse=["csr"], copy=copy,
+                        dtype=[np.float64, np.float32])
 
         n_samples, n_features = X.shape
 
         if self.sublinear_tf:
-            np.log(X.data, X.data)
-            X.data += 1
+            if sp.issparse(X):
```

Comment:
probably. I have not added any new tests, there are still enough existing tests failing. Still wip, mostly wanted feedback on the mixins.